src/apic-extension/azext_apic_extension/command_patches.py

```diff
@@ -582,25 +582,33 @@ def _build_arguments_schema(cls, *args, **kwargs):
             required=True
         )
 
+        args_schema.apim_apis = AAZListArg(
+            options=["--apim-apis"],
+            help="The APIs to be imported.",
+            required=True
+        )
+        args_schema.apim_apis.Element = AAZStrArg()
+
         return args_schema
 
     def pre_operations(self):
         # Set apim_resource_id based on user input
         super().pre_operations()
         args = self.ctx.args
 
-        if not is_valid_resource_id(args.azure_apim.to_serialized_data()):
-            # The APIM is in the same resource group
-            resource_group = args.resource_group
-            subscription_id = self.ctx.subscription_id
-            apim_resource_id = (f"/subscriptions/{subscription_id}/resourceGroups/{resource_group}/providers/"
-                                f"Microsoft.ApiManagement/service/{args.azure_apim}")
-        else:
-            apim_resource_id = args.azure_apim
+        resource_group = args.resource_group
```

Comment:
can you share why we add these logic and remove the old one?

Reply:
actually, the logic should be the current one. I don't know why the correct logic isn't in this branch, so i modified it